crates/subspace-networking/src/constructor.rs

```diff
@@ -469,6 +469,8 @@ where
         gossipsub,
         record_store: LocalOnlyRecordStore::new(local_records_provider),
         request_response_protocols,
+        request_response_max_concurrent_streams: max_established_incoming_connections as usize
```

Comment:
This is a great improvement already, however, shouldn't we multiply the value by the `max_streams_per_connections`?

Reply:
Good point, applied suggestion (also changed commit description/PR title because it didn't make any sense)

Reply:
Actually, I meant `const MAX_CONCURRENT_STREAMS_PER_CONNECTION: usize = 10;` parameters which is different for each behaviour. Usually, it's 10 but Kademlia has 30.

I would set at least 10 as our multiplier.

Reply:
The limit will not be enforced per connection in practice, but let's see how it works out in practice